From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LuneOS / webOS Ports <team@webos-ports.org>
Date: Thu, 13 Aug 2026 00:00:00 +0000
Subject: [PATCH] presageDemo: pass a format string to mvprintw/mvwprintw

Five call sites pass a std::string's c_str() straight as the format argument,
which -Werror=format-security rejects:

  presageDemo.cpp:176:21: error: format not a string literal and no format
  arguments [-Werror=format-security]

OE enables format-security hardening by default, and it is fatal here. Pass
"%s" explicitly rather than turning the check off for the whole recipe.

Upstream-Status: Pending
---
 src/tools/presageDemo.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/tools/presageDemo.cpp b/src/tools/presageDemo.cpp
index 5cfd55d..c63f908 100644
--- a/src/tools/presageDemo.cpp
+++ b/src/tools/presageDemo.cpp
@@ -173,7 +173,7 @@ int main(int argc, char** argv)
 	    // key corresponding to desired token. selecting
 	    // suggestion.
 	    std::string message = "Last selected word: " + words[c - KEY_F0 - 1];
-	    mvprintw(LINES - 3, 0, message.c_str());
+	    mvprintw(LINES - 3, 0, "%s", message.c_str());
             clrtoeol();
 	    move(LINES, COLS);
 
@@ -213,7 +213,7 @@ void draw_context_win(WINDOW* win, std::string str)
 {
     wclear( win );
     box( win, 0, 0 );
-    mvwprintw( win, 1, 1, str.c_str() );
+    mvwprintw( win, 1, 1, "%s", str.c_str());
     wrefresh( win );
 }
 
@@ -226,7 +226,7 @@ void drawMsgWin( WINDOW* win, std::vector<std::string> words )
     int i = 1;
     std::vector<std::string>::const_iterator j = words.begin();
     while( j != words.end() ) {
-	mvwprintw( win, i, 1, j->c_str() );
+	mvwprintw( win, i, 1, "%s", j->c_str());
 	i++;
 	j++;
     }
@@ -241,7 +241,7 @@ void draw_function_keys(WINDOW* win)
     for (int i = 1; i <= atoi(suggestions.c_str()); i++) {
         std::stringstream ss;
         ss << 'F' << i;
-        mvwprintw(win, i, 1, ss.str().c_str());
+        mvwprintw(win, i, 1, "%s", ss.str().c_str());
     }
     wrefresh(win);
 }
@@ -291,7 +291,7 @@ void draw_previous_suggestions(std::vector<std::string> words, bool contextChang
 	     strit != listit->end();
 	     strit++) {
 	    
-	    mvwprintw(win, line, 1, strit->c_str());
+	    mvwprintw(win, line, 1, "%s", strit->c_str());
 	    line++;
 	}
 
